server/chat-plugins/chatlog.ts

```diff
@@ -166,10 +165,89 @@ export const LogReader = new class {
 		const prevMonth = new Date(new Date(`${month}-15`).getTime() - 30 * DAY);
 		return prevMonth.toISOString().slice(0, 7);
 	}
-
 	today() {
 		return Chat.toTimestamp(new Date()).slice(0, 10);
 	}
+	isMonth(text: string) {
+		return /[0-9]{4}-[0-9]{2}/.test(text);
+	}
+	isDay(text: string) {
+		return /[0-9]{4}-[0-9]{2}-[0-9]{2}/.test(text);
+	}
+	async findBattleLog(tier: ID, number: number): Promise<string[] | null> {
+		// binary search!
+		const months = (await FS('logs').readdir()).filter(this.isMonth).sort();
+		if (!months.length) return null;
+
+		// find first day
+		let firstDay!: string;
+		while (months.length) {
+			const month = months[0];
+			if (await FS(`logs/${month}/${tier}`).exists()) {
+				const days = (await FS(`logs/${month}/${tier}/`).readdir()).filter(this.isDay).sort();
+				firstDay = days[0];
+				break;
+			}
+			months.shift();
+		}
+		if (!firstDay) return null;
+
+		// find last day
+		let lastDay!: string;
+		while (months.length) {
+			const month = months[months.length - 1];
+			if (await FS(`logs/${month}/${tier}`).exists()) {
+				const days = (await FS(`logs/${month}/${tier}/`).readdir()).filter(this.isDay).sort();
+				lastDay = days[days.length - 1];
+				break;
+			}
```

Comment:
tbh existence checks are often unnecessary, and this shows why

```suggestion
			try {
				const days = (await FS(`logs/${month}/${tier}/`).readdir()).filter(this.isDay).sort();
				lastDay = days[days.length - 1];
				break;
			} catch (err) {}
```